# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1371738205 14400
#      Thu Jun 20 10:23:25 2013 -0400
# Node ID 0fabe0f49f38f19e1407068f33869309cfb61cf3
# Parent  0bf1d559b3214eb77f55bf383c836583e2714517
use better method to determine whether we are using flex and bison

* m4/acinclude.m4 (OCTAVE_PROG_BISON, OCTAVE_PROG_FLEX): Use --version
output to checkf for bison and flex instead of relying on program
names.

diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -1394,18 +1394,23 @@ AC_DEFUN([OCTAVE_PROG_AR], [
   fi
   AC_SUBST(ARFLAGS)
 ])
 dnl
 dnl Check for bison.
 dnl
 AC_DEFUN([OCTAVE_PROG_BISON], [
   AC_PROG_YACC
-  case "$YACC" in
-    bison*)
+
+  case "`$YACC --version`" in
+    *bison*) tmp_have_bison="yes" ;;
+    *) tmp_have_bison=no ;;
+  esac
+
+  if test "$tmp_have_bison" = yes; then
     AC_CACHE_CHECK([syntax of bison push/pull declaration],
                    [octave_cv_bison_push_pull_decl_style], [
       style="dash underscore"
       quote="noquote quote"
       for s in $style; do
         for q in $quote; do
           if test $s = "dash"; then
             def="%define api.push-pull"
@@ -1435,45 +1440,40 @@ EOF
           fi
         done
         if test $ac_status -eq 0; then
           break
         fi
       done
       rm -f conftest.yy y.tab.h y.tab.c
       ])
-    ;;
-  esac
+  fi
 
   AC_SUBST(BISON_PUSH_PULL_DECL_STYLE, $octave_cv_bison_push_pull_decl_style)
 
   if test -z "$octave_cv_bison_push_pull_decl_style"; then
     YACC=
     warn_bison_push_pull_decl_style="
 
 I wasn't able to find a suitable style for declaring a push-pull
 parser in a bison input file so I'm disabling bison.
 "
     OCTAVE_CONFIGURE_WARNING([warn_bison_push_pull_decl_style])
   fi
 
-  case "$YACC" in
-    bison*)
-    ;;
-    *)
-      YACC='$(top_srcdir)/build-aux/missing bison'
-      warn_bison="
+  if test "$tmp_have_bison" = no; then
+    YACC='$(top_srcdir)/build-aux/missing bison'
+    warn_bison="
 
 I didn't find bison, but it's only a problem if you need to
 reconstruct parse.cc, which is the case if you're building from VCS
 sources.
 "
-      OCTAVE_CONFIGURE_WARNING([warn_bison])
-    ;;
-  esac
+    OCTAVE_CONFIGURE_WARNING([warn_bison])
+  fi
 ])
 dnl
 dnl Find desktop-file-install program.
 dnl
 AC_DEFUN([OCTAVE_PROG_DESKTOP_FILE_INSTALL], [
   AC_CHECK_PROG(DESKTOP_FILE_INSTALL, desktop-file-install, desktop-file-install, [])
   AC_SUBST(DESKTOP_FILE_INSTALL)
 ])
@@ -1492,18 +1492,18 @@ dnl Check for flex.
 dnl
 AC_DEFUN([OCTAVE_PROG_FLEX], [
   ## For now, don't define LEXLIB to be -lfl -- we don't use anything in
   ## it, and it might not be installed.
   ##
   ## Also make sure that we generate an interactive scanner if we are
   ## using flex.
   AC_PROG_LEX
-  case "$LEX" in
-    flex*)
+  case "`$LEX --version`" in
+    *flex*)
       LFLAGS="-I"
       AC_MSG_RESULT([defining LFLAGS to be $LFLAGS])
       LEXLIB=
     ;;
     *)
       LEX='$(top_srcdir)/build-aux/missing flex'
       warn_flex="
 
